### PR TITLE
ci: change integration test profile to be activated based on property

### PR DESCRIPTION
**Pull request by matt-whiteman**
## Description

Renames the 'integration-tests' profile to 'integration-tests-build'

Changes the 'integration-tests-build' profile to use property-based activation instad of activeByDefault. This allows for other maven profiles to be specified without automatic deactivation of this profile.

Logic is: profile is active unless skipQaBuilds property is provided. Behavior was tested to match how it was before.

Relevant slack discussions:

https://camunda.slack.com/archives/C06HTSPD5AP/p1714044873531379
https://camunda.slack.com/archives/C06HTSPD5AP/p1714045046184779

## Related issues

closes #

### CODE REVIEW

File: .github/workflows/camunda-platform-release.yml
Comment (tmetzke) [inline .github/workflows/camunda-platform-release.yml:184]: Do we have to add the `-DskipQaBuild` here inside of the `-Darguments` string as well?

**Review by tmetzke — APPROVED** [review]
Pre-approving as it looks good in general. I added an open question but it's blocking the merge in my view.

File: .github/workflows/camunda-platform-release.yml
Comment (Zelldon) [reply .github/workflows/camunda-platform-release.yml:184]: Yes we do. That is the part which is executed as part of the deploy during release:perform

Comment (matt-whiteman) [reply .github/workflows/camunda-platform-release.yml:184]: This is the same PR as before, I just had to remake it. Getting all sorts of CI issues after rebasing and it seemed simpler to just remake the PR and make sure I didn't mess up anything

Comment (matt-whiteman) [reply .github/workflows/camunda-platform-release.yml:184]: Oh I see what you're saying, I need to move -DskipQaBuild inside the -Darguments, I'll update that now

**Comment by matt-whiteman** [comment]
@tmetzke @Zelldon I added the skipQaBuild inside -Darguments as well. I'll wait to merge this until tomorrow if you want to doublecheck it.

File: .github/workflows/camunda-platform-release.yml
Comment (matt-whiteman) [reply .github/workflows/camunda-platform-release.yml]: I also made a small tweak to make it more consistent with other environment variables so it's clearer for the user.

If the property is not present: QA build occurs (profile is active by default basically)
If the property is present with a value of 'true': QA build occurs (-DskipQaBuild=true)
If the property is present with a value of 'false': QA build does not occur (-DskipQaBuild=false)

This will follow the same behavior as other variables like skipTests.

Comment (tmetzke) [reply .github/workflows/camunda-platform-release.yml]: > If the property is not present: QA build occurs (profile is active by default basically)
If the property is present with a value of 'true': QA build occurs (-DskipQaBuild=true)
If the property is present with a value of 'false': QA build does not occur (-DskipQaBuild=false)

Just to make sure I am not going mad, shouldn't this be:
If the property is not present: QA build occurs (profile is active by default basically)
If the property is present with **any other value than** 'true': QA build occurs (-DskipQaBuild=**false**)
If the property is present with a value of **'true'**: QA build does not occur (-DskipQaBuild=**true**)

Comment (matt-whiteman) [reply .github/workflows/camunda-platform-release.yml]: You are correct, this is what happens when I try to write up explanations when I'm doing too many things at once. I just tested to make sure.

No property, qa build.
Property false, qa build.
Property true, qa build

Sorry for the confusion and churn on this PR.

**Review by tmetzke — APPROVED** [review]
🚀

**Review by ChrisKujawa — APPROVED** [review]
🚀 Thanks 👍🏼

**Pull request merged**
